src/core/SIP/SIPMessageBase.cs

```diff
@@ -37,7 +40,40 @@ public class SIPMessageBase
         /// <summary>
         /// The optional body or payload for the SIP request/response.
         /// </summary>
-        public string Body;
+        public string Body
+        {
+            get
+            {
+                if (_body != null)
+                {
+                    return Encoding.UTF8.GetString(_body);
+                }
+                else
+                {
+                    return null;
+                }
+            }
+            set
+            {
+                if(value == null)
+                {
+                    _body = null;
+                }
+                else
+                {
+                    _body = Encoding.UTF8.GetBytes(value);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Used for binary data payloads where body string might be corrupt due to it's utf-8 encoding.
+        /// </summary>
+        public byte[] RawBuffer

```

Comment:
Since this contains only the Body, should the property be called BodyRaw, BodyBuffer, or BodyRawBuffer?

Reply:
Yes, good point.